src/views/Quiz.vue

```diff
@@ -1,9 +1,121 @@
 <template>
-  <div>Quiz</div>
+  <div class="quiz">
+    <QuizStart
+      v-if="step === 0"
+      :go-to-next-step="goToNextStep"
+    />
+    <form
+      v-else-if="isQuestionStep"
+      class="quiz__form"
+    >
+      <QuizQuestion
+        :question="questions[step-1]"
+        :step="step"
+        :go-to-next-step="goToNextStep"
+      >
+        <div
+          slot="wizard"
+          class="quiz__wizard"
+        >
+          <span class="quiz__wizard-text">{{ questionsLeft }} left</span>
+          <div
+            class="quiz__wizard--inner"
+            :style="innerWizardWidth"
+          />
+        </div>
+      </QuizQuestion>
+    </form>
+    <QuizResult
+      v-else
+      :result="result"
+      :go-to-start="goToStart"
+    />
+  </div>
 </template>
 
 <script>
+import QuizStart from '@/components/QuizStart';
+import QuizQuestion from '@/components/QuizQuestion';
+import QuizResult from '@/components/QuizResult';
+import { QUESTIONS } from '@/constants';
+// import { fetchQuestions } from '@/api';
+
 export default {
+  components: {
+    QuizStart,
+    QuizQuestion,
+    QuizResult,
+  },
+
+  data() {
+    return {
+      step: 0,
+      questions: QUESTIONS,
+      answers: [],
+      result: {
+        name: '',
+        img: '',
+      },
+    };
+  },
+
+  computed: {
+    isQuestionStep() {
+      return this.step <= this.questions.length;
+    },
+
+    questionsLeft() {
+      const questionsLeftCount = this.questions.length - this.step;
+      const questionsLeftPhrase = questionsLeftCount === 1 ? 'question' : 'questions';
+      return `${questionsLeftCount} ${questionsLeftPhrase}`;
+    },
+
+    innerWizardWidth() {
+      const oneStepWidth = 100 / this.questions.length;
+      const width = oneStepWidth * this.step;
+      return {
+        width: `${width}%`,
+      };
+    },
+  },
+
+  methods: {
+    // async loadQuestions() {
+    //   try {
+    //     const { questions } = await fetchQuestions();
+    //     this.questions = questions || [];
+    //   } catch (error) {
+    //     this.questions = [];
+    //     console.error(error);
+    //   }
+    // },
+
+    goToNextStep(answer) {
+      if (this.step === 0) {
+        this.step += 1;
+      } else if (this.step < this.questions.length) {
+        this.answers = [...this.answers, answer];
```

Comment:
As for the answers, wouldn't it be a bit more clear if we set them when answer is passed to the function? That is:
```js
this.step += 1;

if (answer) {
  this.answers = [...this.answers, answer];
}

// if is last answer (computed property) -> submit
```